app/parsers/bulkrax/oai_adventist_qdc_parser.rb

```diff
@@ -1,132 +1,216 @@
 # frozen_string_literal: true
 
 module Bulkrax
+  # **Assumption 1 regarding Adventist's OAI feed:** The works, collections, and file sets are declared
+  # at the OAI record level data.
+  #
+  # **Assumption 2 regarding Adventist's OAI feed:** The operator will run the OAI's set(s) that
+  # contain collections before those collections are declared/referenced in other works.  Due to the
+  # asynchronous nature of the entry jobs this means that an importer job (or jobs) should be run
+  # for sets that define the collections; then the importer job(s) should be run for works and file
+  # sets.  If in the future we find that there is a mixture of collection, work, and file set
+  # objects with an OAI feed we might want to consider parameterizing the importer to say
+  # "collections" or not "collections" and then add conditionals accordingly.
+  #
+  # From these assumptions we need to be mindful that our collection creation is different than
+  # other OAI collection creations.
   class OaiAdventistQdcParser < OaiQualifiedDcParser
     def entry_class
       Bulkrax::OaiAdventistQdcEntry
     end
 
+    alias work_entry_class entry_class
+
     def collection_entry_class
-      OaiSetEntry
+      Bulkrax::OaiAdventistSetEntry
     end
 
     def file_set_entry_class
       RdfFileSetEntry
     end
 
+    # Setting both #count_towards_limit and #counters as part of initialization so that regardless
+    # of how we create the objects (e.g. either sequentially calling #create_collections,
+    # #create_file_sets, #create_works, #create_relationships OR calling #create_objects) we
+    # adequately handle the various counts.
+    def initialize(*args, &block)
+      super
+      @count_towards_limit = 0
+      @counters = { file_set: 0, collection: 0, work: 0 }
+      @model_field_mappings = Bulkrax.field_mappings[self.class.to_s]&.dig('model', :from) || ["model"]
```

Comment:
This is an interesting behavioral change. Normally, the field mapping are saved with the importer. That means importers do not change field mapping if the yaml is updated later. This seems to take the current field mapping at time of each run. Is that intentional? I can give more color on what field mappings are saved if you need that

Reply:
It is partially intentional, but is most definitely something I want to leverage. I referenced the [`Bulkrax::ApplicationParser#model_field_mappings`](https://github.com/samvera-labs/bulkrax/blob/b857a25becefc4bd960f25647033dfb2e899386e/app/parsers/bulkrax/application_parser.rb#L120-L125); and in hindsight I can remove that.  (Though I'm now curious why I chose to include that).

```ruby
    def model_field_mappings
      model_mappings = Bulkrax.field_mappings[self.class.to_s]&.dig('model', :from) || []
      model_mappings |= ['model']

      model_mappings
    end
```

The above code is somewhat similar (though there is a difference).  My intention in adding this instance variable was to cache the value because later one we are continually calling the `#model_field_mappings`.

So there are three changes:

1. Restore the logic of `model_field_mappings` so that we have the `model_mappings |= ['model']` behavior.
2. Cache the value for the object's instantiation.
3. When we instantiate the object favor the persisted mappings that are part of the Entry/Importer (instead of derived from the bulkrax configuration)

It is item 3 that I believe is what you are referencing in regards to "more color on what field mappings are saved".  I can resolve item 1 by removing the line from the object.  And item 2 I can look at updating in Bulkrax.